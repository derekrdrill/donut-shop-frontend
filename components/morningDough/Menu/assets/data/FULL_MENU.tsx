import avocadoToast from '../images/bread/avocadoToast.png';
import blueberryMuffin from '../images/bread/blueberryMuffin.png';
import chocolateMuffin from '../images/bread/chocolateMuffin.png';
import croissant from '../images/bread/croissant.png';
import everythingBagel from '../images/bread/everythingBagel.png';
import plainBagel from '../images/bread/plainBagel.png';
import poppyBagel from '../images/bread/poppyBagel.png';
import raisinBagel from '../images/bread/raisinBagel.png';
import cocaCola from '../images/coffeeAndTea/cocaCola.png';
import hotCoffee from '../images/coffeeAndTea/hotCoffee.png';
import hotLatte from '../images/coffeeAndTea/hotLatte.png';
import hotTea from '../images/coffeeAndTea/hotTea.png';
import icedCoffee from '../images/coffeeAndTea/icedCoffee.png';
import icedTea from '../images/coffeeAndTea/icedTea.png';
import sprite from '../images/coffeeAndTea/sprite.png';
import waterBottle from '../images/coffeeAndTea/waterBottle.png';
import bostonCreamDonut from '../images/donuts/bostonCreamDonut.png';
import chocolateDonut from '../images/donuts/chocolateDonut.png';
import chocolateFrostingDonut from '../images/donuts/chocolateFrostingDonut.png';
import chocolateFrostingSprinklesDonut from '../images/donuts/chocolateFrostingSprinklesDonut.png';
import glazedDonut from '../images/donuts/glazedDonut.png';
import jellyFilledDonut from '../images/donuts/jellyFilledDonut.png';
import powderedDonut from '../images/donuts/powderedDonut.png';
import sprinkleDonut from '../images/donuts/sprinkleDonut.png';

export interface FullMenuItem {
  key: string;
  name: string;
  image: string;
  category: string;
  subCategory: string;
  soda?: boolean;
  bottled?: boolean;
}

const FULL_MENU: FullMenuItem[] = [
  {
    key: 'chocolateDonut',
    name: 'Chocolate Donut',
    image: chocolateDonut,
    category: 'donuts',
    subCategory: 'donuts',
  },
  {
    key: 'glazedDonut',
    name: 'Glazed Donut',
    image: glazedDonut,
    category: 'donuts',
    subCategory: 'donuts',
  },
  {
    key: 'powderedDonut',
    name: 'Powdered Donut',
    image: powderedDonut,
    category: 'donuts',
    subCategory: 'donuts',
  },
  {
    key: 'sprinkleDonut',
    name: 'Strawberry sprinkles donut',
    image: sprinkleDonut,
    category: 'donuts',
    subCategory: 'donuts',
  },
  {
    key: 'chocolateFrostingDonut',
    name: 'Chocolate frosting donut',
    image: chocolateFrostingDonut,
    category: 'donuts',
    subCategory: 'donuts',
  },
  {
    key: 'chocolateFrostingSprinklesDonut',
    name: 'Chocolate w/ sprinkles donut',
    image: chocolateFrostingSprinklesDonut,
    category: 'donuts',
    subCategory: 'donuts',
  },
  {
    key: 'jellyFilledDonut',
    name: 'Jelly Filled Donut',
    image: jellyFilledDonut,
    category: 'donuts',
    subCategory: 'donuts',
  },
  {
    key: 'bostonCreamDonut',
    name: 'Boston Cream Donut',
    image: bostonCreamDonut,
    category: 'donuts',
    subCategory: 'donuts',
  },
  {
    key: 'plainBagel',
    name: 'Plain Bagel',
    image: plainBagel,
    category: 'bread',
    subCategory: 'bagels',
  },
  {
    key: 'poppyBagel',
    name: 'Poppy Bagel',
    image: poppyBagel,
    category: 'bread',
    subCategory: 'bagels',
  },
  {
    key: 'everythingBagel',
    name: 'Everything Bagel',
    image: everythingBagel,
    category: 'bread',
    subCategory: 'bagels',
  },
  {
    key: 'raisinBagel',
    name: 'Raisin Bagel',
    image: raisinBagel,
    category: 'bread',
    subCategory: 'bagels',
  },
  {
    key: 'blueberryMuffin',
    name: 'Blueberry Muffin',
    image: blueberryMuffin,
    category: 'bread',
    subCategory: 'muffins',
  },
  {
    key: 'chocolateMuffin',
    name: 'Chocolate Muffin',
    image: chocolateMuffin,
    category: 'bread',
    subCategory: 'muffins',
  },
  {
    key: 'croissant',
    name: 'Croissant',
    image: croissant,
    category: 'bread',
    subCategory: 'croissants',
  },
  {
    key: 'avocadoToast',
    name: 'Avocado Toast',
    image: avocadoToast,
    category: 'bread',
    subCategory: 'toast',
  },
  {
    key: 'hotCoffee',
    name: 'Hot Coffee',
    image: hotCoffee,
    category: 'drink',
    subCategory: 'hotDrinks',
  },
  {
    key: 'icedCoffee',
    name: 'Iced Coffee',
    image: icedCoffee,
    category: 'drink',
    subCategory: 'coldDrinks',
  },
  {
    key: 'hotLatte',
    name: 'Hot Latte',
    image: hotLatte,
    category: 'drink',
    subCategory: 'hotDrinks',
  },
  { key: 'hotTea', name: 'Hot Tea', image: hotTea, category: 'drink', subCategory: 'hotDrinks' },
  {
    key: 'icedTea',
    name: 'Iced Tea',
    image: icedTea,
    category: 'drink',
    subCategory: 'coldDrinks',
    soda: true,
  },
  {
    key: 'cocaCola',
    name: 'Coca-Cola',
    image: cocaCola,
    category: 'drink',
    subCategory: 'coldDrinks',
    soda: true,
  },
  {
    key: 'sprite',
    name: 'Sprite',
    image: sprite,
    category: 'drink',
    subCategory: 'coldDrinks',
    soda: true,
  },
  {
    key: 'waterBottle',
    name: 'Water Bottle',
    image: waterBottle,
    category: 'drink',
    subCategory: 'coldDrinks',
    bottled: true,
  },
];

export default FULL_MENU;
