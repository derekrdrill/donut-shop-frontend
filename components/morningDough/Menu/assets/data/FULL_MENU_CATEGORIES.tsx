import bagels from '../images/bread/plainBagel.png';
import croissants from '../images/bread/croissant.png';
import donuts from '../images/donuts/chocolateFrostingDonut.png';
import hotDrinks from '../images/coffeeAndTea/hotLatte.png';
import coldDrinks from '../images/coffeeAndTea/icedCoffee.png';
import muffins from '../images/bread/blueberryMuffin.png';
import toast from '../images/bread/avocadoToast.png';

export interface FullMenuCategory {
  key: string;
  name: string;
  image: string;
}

const FULL_MENU_CATEGORIES: FullMenuCategory[] = [
  { key: 'hotDrinks', name: 'Hot Drinks', image: hotDrinks },
  { key: 'coldDrinks', name: 'Cold Drinks', image: coldDrinks },
  { key: 'donuts', name: 'Donuts', image: donuts },
  { key: 'bagels', name: 'Bagels', image: bagels },
  { key: 'muffins', name: 'Muffins', image: muffins },
  { key: 'croissants', name: 'Croissants', image: croissants },
  { key: 'toast', name: 'Toast', image: toast },
];

export default FULL_MENU_CATEGORIES;
