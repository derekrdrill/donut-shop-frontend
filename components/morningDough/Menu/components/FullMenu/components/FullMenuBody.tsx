import * as React from 'react';
import { useRouter } from 'next/router';
import { Grid, Typography } from '@mui/material';

import FullMenuBackButton from './FullMenuBackButton';

import {
  FullMenuContainer,
  FullMenuItemRoot,
  FullMenuItemContainer,
  FullMenuItemImage,
} from '../style';

import { FullMenuCategory } from '../../../assets/data/FULL_MENU_CATEGORIES';
import { FullMenuItem } from '../../../assets/data/FULL_MENU';

export const setFullMenuDataHelper = (menuData: FullMenuItem[], selectedCategory: string) =>
  menuData.filter(menuItem => menuItem.subCategory === selectedCategory);

interface FullMenuBodyProps {
  fullMenuCategories: FullMenuCategory[];
  fullMenu: FullMenuItem[];
}

const FullMenuBody = ({ fullMenuCategories, fullMenu }: FullMenuBodyProps) => {
  const router = useRouter();

  const [fullMenuDataClone] = React.useState<FullMenuItem[]>(fullMenu);
  const [fullMenuCategoriesClone] = React.useState<FullMenuCategory[]>(fullMenuCategories);

  const [fullMenuData, setFullMenuData] = React.useState<FullMenuCategory[] | FullMenuItem[]>(
    fullMenuCategories,
  );

  const [selectedCategory, setSelectedCategory] = React.useState<string | null>(null);
  const [selectedCategoryName, setSelectedCategoryName] = React.useState<string | null>(null);

  React.useEffect(() => {
    if (!selectedCategory) {
      setFullMenuData(fullMenuCategoriesClone);
    } else {
      setFullMenuData(setFullMenuDataHelper(fullMenuDataClone, selectedCategory));
    }
  }, [selectedCategory]);

  return (
    <FullMenuContainer data-testid='full-menu' container rowGap={3}>
      <Grid item xs={12}>
        <Typography textAlign='center' variant='h4'>
          {selectedCategoryName ?? 'Full Menu'}
        </Typography>
        <FullMenuBackButton
          selectedCategory={selectedCategory}
          setSelectedCategory={setSelectedCategory}
          setSelectedCategoryName={setSelectedCategoryName}
        />
      </Grid>
      {fullMenuData.map(menuItem => (
        <FullMenuItemRoot key={menuItem.key} item xs={6} md={4} xl={3}>
          <FullMenuItemContainer
            container
            onClick={
              /* istanbul ignore next */
              () => {
                if (!selectedCategory) {
                  setSelectedCategory(menuItem.key);
                  setSelectedCategoryName(menuItem.name);
                } else {
                  router.push(`/menu/${selectedCategory}/${menuItem.key}`);
                }
              }
            }
          >
            <Grid item xs={12}>
              <Grid container justifyContent='center'>
                <FullMenuItemImage src={menuItem.image} />
              </Grid>
            </Grid>
            <Grid item xs={12}>
              <Typography variant='h6' textAlign='center' sx={{ paddingBottom: '10px' }}>
                {menuItem.name}
              </Typography>
            </Grid>
          </FullMenuItemContainer>
        </FullMenuItemRoot>
      ))}
    </FullMenuContainer>
  );
};

export default FullMenuBody;
