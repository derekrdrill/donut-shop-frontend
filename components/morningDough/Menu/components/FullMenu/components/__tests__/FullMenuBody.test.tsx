jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => jest.fn(),
}));

import '@testing-library/jest-dom';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import FullMenuBody, { setFullMenuDataHelper } from '../FullMenuBody';

import FULL_MENU from '../../../../assets/data/FULL_MENU';
import FULL_MENU_CATEGORIES from '../../../../assets/data/FULL_MENU_CATEGORIES';

describe('FullMenuBody unit tests', () => {
  it('renders as expected', () => {
    const result = renderer
      .create(<FullMenuBody fullMenuCategories={FULL_MENU_CATEGORIES} fullMenu={FULL_MENU} />)
      .toJSON();
    expect(result).toMatchSnapshot();
  });

  it('renders FullMenuBody when category is selected', async () => {
    render(<FullMenuBody fullMenuCategories={FULL_MENU_CATEGORIES} fullMenu={FULL_MENU} />);

    jest
      .spyOn(React, 'useState')
      .mockImplementationOnce(() => [[], () => null])
      .mockImplementationOnce(() => [[], () => null])
      .mockImplementationOnce(() => [[], () => null])
      .mockImplementationOnce(() => ['hotDrinks', () => null]);

    const fullMenuBody = await screen.findAllByTestId('full-menu');

    console.log(fullMenuBody[0]);

    // expect(fullMenuBody[0]).toHaveClass('selected-category');
  });

  it('runs setFullMenuDataHelper', () => {
    const data = [
      { key: '1', name: 'coffee', image: 'coffee.png', category: 'drinks', subCategory: 'coffee' },
      { key: '2', name: 'tea', image: 'tea.png', category: 'drinks', subCategory: 'tea' },
      { key: '3', name: 'icedTea', image: 'icedTea.png', category: 'drinks', subCategory: 'tea' },
    ];

    expect(setFullMenuDataHelper(data, 'tea')).toEqual([
      { key: '2', name: 'tea', image: 'tea.png', category: 'drinks', subCategory: 'tea' },
      { key: '3', name: 'icedTea', image: 'icedTea.png', category: 'drinks', subCategory: 'tea' },
    ]);

    expect(setFullMenuDataHelper(data, 'coffee')).toEqual([
      { key: '1', name: 'coffee', image: 'coffee.png', category: 'drinks', subCategory: 'coffee' },
    ]);
  });
});
