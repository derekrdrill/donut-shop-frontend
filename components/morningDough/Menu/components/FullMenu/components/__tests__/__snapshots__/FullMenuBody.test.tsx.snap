// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullMenuBody unit tests renders as expected 1`] = `
<div
  className="MuiGrid-root MuiGrid-container sc-dmqHEX hMzWU css-851zx1-MuiGrid-root"
  data-testid="full-menu"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
  >
    <h4
      className="MuiTypography-root MuiTypography-h4 css-149xwtl-MuiTypography-root"
    >
      Full Menu
    </h4>
    <div
      className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
      data-testid="full-menu-back-button"
    />
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-4 MuiGrid-grid-xl-3 sc-jTrPJq dulzkK css-9vv5x2-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container sc-gLDzan kAAEIz css-11lq3yg-MuiGrid-root"
      onClick={[Function]}
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
        >
          <img
            className="sc-iAEyYk lhGoHL"
            src={
              {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 css-14bvv6x-MuiTypography-root"
        >
          Hot Drinks
        </h6>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-4 MuiGrid-grid-xl-3 sc-jTrPJq dulzkK css-9vv5x2-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container sc-gLDzan kAAEIz css-11lq3yg-MuiGrid-root"
      onClick={[Function]}
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
        >
          <img
            className="sc-iAEyYk lhGoHL"
            src={
              {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 css-14bvv6x-MuiTypography-root"
        >
          Cold Drinks
        </h6>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-4 MuiGrid-grid-xl-3 sc-jTrPJq dulzkK css-9vv5x2-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container sc-gLDzan kAAEIz css-11lq3yg-MuiGrid-root"
      onClick={[Function]}
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
        >
          <img
            className="sc-iAEyYk lhGoHL"
            src={
              {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 css-14bvv6x-MuiTypography-root"
        >
          Donuts
        </h6>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-4 MuiGrid-grid-xl-3 sc-jTrPJq dulzkK css-9vv5x2-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container sc-gLDzan kAAEIz css-11lq3yg-MuiGrid-root"
      onClick={[Function]}
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
        >
          <img
            className="sc-iAEyYk lhGoHL"
            src={
              {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 css-14bvv6x-MuiTypography-root"
        >
          Bagels
        </h6>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-4 MuiGrid-grid-xl-3 sc-jTrPJq dulzkK css-9vv5x2-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container sc-gLDzan kAAEIz css-11lq3yg-MuiGrid-root"
      onClick={[Function]}
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
        >
          <img
            className="sc-iAEyYk lhGoHL"
            src={
              {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 css-14bvv6x-MuiTypography-root"
        >
          Muffins
        </h6>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-4 MuiGrid-grid-xl-3 sc-jTrPJq dulzkK css-9vv5x2-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container sc-gLDzan kAAEIz css-11lq3yg-MuiGrid-root"
      onClick={[Function]}
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
        >
          <img
            className="sc-iAEyYk lhGoHL"
            src={
              {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 css-14bvv6x-MuiTypography-root"
        >
          Croissants
        </h6>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-4 MuiGrid-grid-xl-3 sc-jTrPJq dulzkK css-9vv5x2-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container sc-gLDzan kAAEIz css-11lq3yg-MuiGrid-root"
      onClick={[Function]}
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
        >
          <img
            className="sc-iAEyYk lhGoHL"
            src={
              {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 css-14bvv6x-MuiTypography-root"
        >
          Toast
        </h6>
      </div>
    </div>
  </div>
</div>
`;
